finagle-core/src/test/scala/com/twitter/finagle/service/StatsFilterTest.scala

```diff
@@ -141,6 +156,17 @@ class StatsFilterTest extends FunSuite {
     assert(receiver.gauges(Seq("pending"))() == 0.0)
   }
 
+  test("don't report pending requests after uncaught exceptions") {
+    val (_, receiver, statsService) = getService(behaviour = Fail)
+    assert(receiver.gauges(Seq("pending"))() == 0.0)
+
+    intercept[ControlThrowable] {
+      statsService("foo")
```

Comment:
it would be nice to validate that pending actually increments.  how about adding a new filter in between StatsFilter and the service that throws the exception that checks that pending is 1.0?